core/templates/dev/head/App.ts

```diff
@@ -103,12 +104,17 @@ oppia.config([
     // Add an interceptor to convert requests to strings and to log and show
     // warnings for error responses.
     $httpProvider.interceptors.push([
-      '$q', '$log', 'AlertsService', function($q, $log, AlertsService) {
+      '$q', '$log', 'AlertsService', 'CsrfTokenService',
+      function($q, $log, AlertsService, CsrfTokenService) {
+        var csrfToken;
+        CsrfTokenService.getToken().then(function(token) {
+          csrfToken = token;
+        });
         return {
           request: function(config) {
             if (config.data) {
               config.data = $.param({
-                csrf_token: GLOBALS.csrf_token,
+                csrf_token: csrfToken,
                 payload: JSON.stringify(config.data),
                 source: document.URL
               }, true);
```

Comment:
This needs to be inside the `.then(function(token) {` too, because if token takes long to load the method will return without loading.